### PR TITLE
Feature/set restrictions

### PR DESCRIPTION
**Pull request by LeoEcker**

### CODE REVIEW

File: CfStreamUploader/CfStreamUploader.Core/Models/AccessRules.cs [hunk 1]
```diff
@@ -1,12 +1,136 @@
...
+        {
+            this.Any = new Any();
+            this.Ip = new Ip();
+            this.Country = new Country();
+        }
+    }
+
+    #region restrictions
+
+    public class Any
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Core/Models/AccessRules.cs:22]: Any, Country, Ip : BaseClass

File: CfStreamUploader/CfStreamUploader.Core/Models/UserSettings.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Text.Json.Serialization;
+
+namespace CfStreamUploader.Core.Models
+{
+    public class UserSettings
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Core/Models/UserSettings.cs:5]: Do we need all constructors?

File: CfStreamUploader/CfStreamUploader.Presentation/App.xaml [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
              xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-             xmlns:local="clr-namespace:CfStreamUploader.Presentation"
-             StartupUri="MainWindow.xaml">
+             xmlns:local="clr-namespace:CfStreamUploader.Presentation.ViewModels"
+             StartupUri="Windows/MainWindow.xaml">
     <Application.Resources>
         <ResourceDictionary>
-            <local:ViewModel x:Key="ViewModel" />
+            <local:MainViewModel x:Key="ViewModel"/>
+            <local:EditViewModel x:Key="EditViewModel"/>
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/App.xaml:9]: Rename: EditRestrictionWindow

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+                FileName = "cmd",
+                WindowStyle = ProcessWindowStyle.Hidden,
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                Arguments = $"/c start {this.currentCodesUrl}"
+            };
+            Process.Start(psi);
+        }
+
+        private void AllowAll()
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs:116]: Allow or Block

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                this.ConfigManager.Config.AccessRules.Any.Allow();
+            }
+            else
+            {
+                this.AnyAction = "block";
+                this.ConfigManager.Config.AccessRules.Any.Block();
+            }
+        }
+
+        private void AllowCountries()
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs:130]: +block

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+                this.ConfigManager.Config.AccessRules.Country.Allow();
+            }
+            else
+            {
+                this.CountryAction = "block";
+                this.ConfigManager.Config.AccessRules.Country.Block();
+            }
+        }
+
+        private void AllowIps()
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs:144]: +block

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/MainViewModel.cs [hunk 7]
```diff
@@ -76,7 +106,38 @@ public ViewModel()
...
-        #region Methods
+        #region public
+
+        public void DragOver(IDropInfo dropInfo)
+        {
+            var dragFileList = ((DataObject) dropInfo.Data).GetFileDropList().Cast<string>();
+            dropInfo.Effects = dragFileList.Any(item =>
+            {
+                var extension = Path.GetExtension(item);
+                return extension != null && extension.Equals(".mp4");
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/MainViewModel.cs:117]: add Issue

File: CfStreamUploader/CfStreamUploader.Presentation/Windows/EditWindow.xaml [hunk 8]
```diff
@@ -0,0 +1,94 @@
+﻿<Window x:Class="CfStreamUploader.Presentation.Windows.EditWindow"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+        mc:Ignorable="d"
+        Title="Edit" Height="300" Width="470"
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/Windows/EditWindow.xaml:7]: Titel = Set restrictions

File: CfStreamUploader/CfStreamUploader.Presentation/Windows/WindowManager.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
+﻿using CfStreamUploader.Presentation.ViewModels;
+
+namespace CfStreamUploader.Presentation.Windows
+{
+    public static class WindowManager
+    {
+        public static EditWindow EditWindow { get; set; }
+
+        public static void OpenEditWindow()
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/Windows/WindowManager.cs:9]: OpenSetRes__Wondow

**Review by TobiasMeissner2000 — APPROVED** [review]

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs (hunk 5 above)
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs:130]: Why only allow?

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs (hunk 6 above)
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditViewModel.cs:144]: SetIpAction*

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/MainViewModel.cs [hunk 10]
```diff
@@ -183,6 +217,17 @@ private void UpdateConfig()
             this.Core.ConfigManager.UpdateConfig(config);
         }
 
+        private void EditRestrictions()
+        {
+            WindowManager.OpenEditWindow();
+
+            this.Core.ConfigManager.ReadConfig();
+
+            this.RestrictionCountry = this.Core.ConfigManager.Config.AccessRules.Country.PrintRestriction();
```
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/MainViewModel.cs:226]: In a privat Void Method?

File: CfStreamUploader/CfStreamUploader.Presentation/Windows/EditWindow.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+
+namespace CfStreamUploader.Presentation.Windows
+{
+    /// <summary>
+    /// Interaktionslogik für EditWindow.xaml
```
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation/Windows/EditWindow.xaml.cs:16]: In english please ;)

**Pull request merged**